### PR TITLE
Titanium and plastitanium shards and weapons + missing textures.

### PR DESCRIPTION
**Pull request by Comxy**
## Why It's Good For The Game
Ever been bothered by why titanium glass and plastitanium glass do not drop their own shard types? Well this is the perfect PR for you! Titanium and plastitanium glass shards never existed, and it is probably because the person who made glass way back in the day didn't have time to add these shards. Luckily I decided to add them after all this time. Every piece of code created has been carefully considered and copied form other code, so then you know it is good code. Also I added more tags, I looked at the guidelines and found that adding the fix and qol tags probably boosts my pr score so it will get merged.
promo1
promo2

## Changelog
- Adds titanium glass shards
- Adds plastitanium glass shards
- Adds plasma glass shiv
- Adds titanium glass shiv
- Adds plastitanium glass shiv
- Adds titanium glass spear
- Adds plastitanium glass spear
- Adds time for the creation of shards in stronger glass types

:cl:
fix: When holding plasma glass, the sprite of normal glass was shown in hand since plasma glass in hand didn't exist.
add: Added titanium and plastitanium glass shards as well as spears and shivs for their types, including plasma glass shivs.
imageadd: Sprites for plasma, titanium and plastitanium glass shards when hold in hand.
qol: titanium and plastitanium glass having their own shards makes sense, and is a quality feature.
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
As someone who's static is digi I am in pain, but I approve nonetheless.

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 1]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+	if(istype(I, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/C = I
+		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
+		if(do_after(user, 70, target = src))
+			var/obj/item/knife/shiv/plasma/S = new /obj/item/knife/shiv/plasma
+			C.use(1)
+			to_chat(user, span_notice("You wrap the [C] around the [src] forming a makeshift weapon."))
+			remove_item_from_storage(src)
+			qdel(src)
+			user.put_in_hands(S)
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:395]: you copy paste this entire thing three times, just make it part of `/obj/item/shard`. Keep it DRY! Don't Repeat Yourself!

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 2]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
 	force = 6
 	throwforce = 11
 	icon_state = "plasmalarge"
+	inhand_icon_state = "shard-plasma"
 	custom_materials = list(/datum/material/alloy/plasmaglass=MINERAL_MATERIAL_AMOUNT)
 	icon_prefix = "plasma"
 	weld_material = /obj/item/stack/sheet/plasmaglass
+
+/obj/item/shard/plasma/attackby(obj/item/I, mob/user, params)
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:385]: a better variable name than `I`, please

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 3]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+	if(istype(I, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/C = I
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:387]: a better variable name than `C`, please

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 4]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
+		if(do_after(user, 70, target = src))
+			var/obj/item/knife/shiv/plasma/S = new /obj/item/knife/shiv/plasma
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:390]: a better variable name than `S`, please

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 5]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+	name = "bright shard"
+	desc = "A nasty looking shard of titanium infused glass."
+	throwforce = 12
+	icon_state = "titaniumlarge"
+	inhand_icon_state = "shard-titanium"
+	custom_materials = list(/datum/material/alloy/titaniumglass=MINERAL_MATERIAL_AMOUNT)
+	icon_prefix = "titanium"
+	weld_material = /obj/item/stack/sheet/titaniumglass
+
+/obj/item/shard/titanium/attackby(obj/item/I, mob/user, params)
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:407]: these have the same issues but you'll just move this into `/obj/item/shard` behavior anyway

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 6]
```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+	desc = "A nasty looking shard of titanium infused plasma glass."
+	force = 7
+	throwforce = 12
+	icon_state = "plastitaniumlarge"
+	inhand_icon_state = "shard-plastitanium"
+	custom_materials = list(/datum/material/alloy/plastitaniumglass=MINERAL_MATERIAL_AMOUNT)
+	icon_prefix = "plastitanium"
+	weld_material = /obj/item/stack/sheet/plastitaniumglass
+
+/obj/item/shard/plastitanium/attackby(obj/item/I, mob/user, params)
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:430]: these have the same issues but you'll just move this into `/obj/item/shard` behavior anyway

**Review by tralezab — CHANGES_REQUESTED** [review]
Codewise, some things need to be cleaned up

**Comment by tralezab** [comment]
On the balance side of things I worry a little bit about reintroducing "spears breaking doors" meta with the plasmashard spear breaking doors when thrown but maybe it's annoying enough to get that it's fine. I dunno if anyone here has tried to break a plasmaglass window but it fuckin blooooows

**Comment by itseasytosee** [comment]
Ungodly

**Comment by TheBonded** [comment]
are titanium spears having 1 higher throwforce than plasma spears (but 1 lower force) intended?

**Comment by thgvr** [comment]
tralezab, we already have spears that can break airlocks. Bamboo spears have 22 throwforce, same as these new ones

**Comment by Comxy** [comment]
I changed some things. I made sure titanium glass spears cannot break doors, and increased some other stats to make it different. Also, you cannot get plastitanium shards anymore by crafting with sheets, since it is too hard to break. Now you can get is by breaking a plastitanium window instead, so it is harder to get. Also I fixed the code up.

**Comment by bruhlookatthisdood** [comment]
i have never seen a "large shard"

File: code/game/objects/items/knives.dm [hunk 7]
```diff
@@ -156,6 +156,49 @@
...
+
+/obj/item/knife/shiv/titanium
+	name = "titanium shiv"
+	icon = 'icons/obj/shards.dmi'
+	icon_state = "titaniumshiv"
+	inhand_icon_state = "titaniumshiv"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	desc = "A makeshift titanium infused glass shiv."
+	force = 8
```
Comment (VladinXXV) [inline code/game/objects/items/knives.dm:180]: Isn't it redundant to have `force = 8` here? It inherits the exact same damage from the base shiv, no? Same stuff for the following in all three of them:
`icon = 'icons/obj/shards.dmi'`
`lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'`
`righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'`

**Review by VladinXXV — COMMENTED** [review]
this pr is swag

**Comment by TheBonded** [comment]
> i have never seen a "large shard"

they dont have any actual differences other than their sprite as far as i know, youve probably seen one and just didnt pay attention since its still just called "Glass Shard" in game

**Comment by Comxy** [comment]
It should be ready to get merged, the code is fine now, the balancing is very good, feel free to merge it any day.

**Comment by TheBonded** [comment]
> It should be ready to get merged, the code is fine now, the balancing is very good, feel free to merge it any day.

image
## when your pr is ready to be re-reviewed and merged:

in the top of the pr, where it shows reviewers, be sure to press the little circle to re-request a review from anyone who requested changes when you are ready for the pr to be checked and merged again.

## that way you dont end up with ugly red circles on your PR like this!
image

**Comment by TheBonded** [comment]
(this is obviously not required for your pr to be merged but it does look nice and notify the people who reviewed that you requested them to come back and review the pr again)

**Comment by TheBonded** [comment]
resolves https://github.com/tgstation/tgstation/issues/66658

**Comment by bruhlookatthisdood** [comment]
(why is the plasma shards slightly darker than the plasma shiv)

**Comment by Comxy** [comment]
> (why is the plasma shards slightly darker than the plasma shiv)

That is probably shading and there are more cracks in the shiv one.

File: code/game/objects/items/knives.dm [hunk 8]
```diff
@@ -156,6 +156,41 @@
...
+	force = 9
+	throwforce = 13
+	armor = list(MELEE = 25, BULLET = 25, LASER = 25, ENERGY = 25, BOMB = 25, BIO = 0, FIRE = 50, ACID = 50)
+	custom_materials = list(/datum/material/glass=400, /datum/material/plasma=200)
+
+/obj/item/knife/shiv/titanium
+	name = "titanium shiv"
+	icon_state = "titaniumshiv"
+	inhand_icon_state = "titaniumshiv"
+	desc = "A makeshift titanium infused glass shiv."
```
Comment (ATH1909) [inline code/game/objects/items/knives.dm:173]:
```suggestion
	desc = "A makeshift titanium-infused glass shiv."
```

File: code/game/objects/items/knives.dm [hunk 9]
```diff
@@ -156,6 +156,41 @@
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 0, ACID = 0)
 	custom_materials = list(/datum/material/glass=400)
 
+/obj/item/knife/shiv/plasma
+	name = "plasma shiv"
+	icon_state = "plasmashiv"
+	inhand_icon_state = "plasmashiv"
+	desc = "A makeshift plasma glass shiv."
```
Comment (ATH1909) [inline code/game/objects/items/knives.dm:163]:
```suggestion
	desc = "A makeshift plasmaglass shiv."
```

File: code/game/objects/items/knives.dm [hunk 10]
```diff
@@ -156,6 +156,41 @@
...
+	throw_range = 7
+	wound_bonus = 10
+	armor = list(MELEE = 25, BULLET = 25, LASER = 25, ENERGY = 25, BOMB = 25, BIO = 0, FIRE = 50, ACID = 50)
+	custom_materials = list(/datum/material/glass=400, /datum/material/titanium=200)
+
+/obj/item/knife/shiv/plastitanium
+	name = "plastitanium shiv"
+	icon_state = "plastitaniumshiv"
+	inhand_icon_state = "plastitaniumshiv"
+	desc = "A makeshift titanium infused plasma glass shiv."
```
Comment (ATH1909) [inline code/game/objects/items/knives.dm:184]:
```suggestion
	desc = "A makeshift titanium-infused plasmaglass shiv."
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 11]
```diff
@@ -77,7 +77,7 @@ GLOBAL_LIST_INIT(glass_recipes, list ( \
 GLOBAL_LIST_INIT(pglass_recipes, list ( \
 	new/datum/stack_recipe("directional window", /obj/structure/window/plasma/unanchored, time = 0, on_floor = TRUE, window_checks = TRUE), \
 	new/datum/stack_recipe("fulltile window", /obj/structure/window/plasma/fulltile/unanchored, 2, time = 0, on_floor = TRUE, window_checks = TRUE), \
-	new/datum/stack_recipe("plasma glass shard", /obj/item/shard/plasma, time = 0, on_floor = TRUE) \
+	new/datum/stack_recipe("plasma glass shard", /obj/item/shard/plasma, time = 20, on_floor = TRUE) \
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/glass.dm:80]:
```suggestion
	new/datum/stack_recipe("plasmaglass shard", /obj/item/shard/plasma, time = 20, on_floor = TRUE) \
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 12]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+		var/obj/item/lightreplacer/lightreplacer = item
+		lightreplacer.attackby(src, user)
+	else if(istype(item, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/cloth = item
+		to_chat(user, span_notice("You begin to wrap the [cloth] around the [src]..."))
+		if(istype(src, /obj/item/shard/plasma))
+			if(do_after(user, 140, target = src))
+				var/obj/item/knife/shiv/plasma/shiv = new /obj/item/knife/shiv/plasma
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/glass.dm:345]:
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 13]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/titanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/titanium/shiv = new /obj/item/knife/shiv/titanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/glass.dm:353]:
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 14]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/plastitanium))
+			if(do_after(user, 70, target = src))
+				var/obj/item/knife/shiv/plastitanium/shiv = new /obj/item/knife/shiv/plastitanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/glass.dm:361]:
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 15]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
+				var/obj/item/knife/shiv/shiv = new /obj/item/knife/shiv
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src] forming a makeshift weapon."))
```
Comment (ATH1909) [inline code/game/objects/items/stacks/sheets/glass.dm:369]:
```suggestion
				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
```

**Comment by Ebb-Real** [comment]
This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.

**Comment by Comxy** [comment]
> This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.

What do you mean with GAGS?

**Comment by Ebb-Real** [comment]
> > This could probably be turned into GAGS if you're up for it. Would maybe be a bit difficult to get the texture and color right though.
> 
> What do you mean with GAGS?

GAGS is a way to assemble a sprite with layers and assigned colors. This means if you have multiple sprites that are just recolors of the original you can turn them into GAGS and just have one or two sprites that is white with shading and assign colors to them with code and it will color it for you, the assigned color's brightness will be adjusted by the shading you added to the GAGS sprite.

This is good because its easy to recolor sprites (in this case if someone adds a new type of glass they just gotta assign the color of the new glass) and because you need less sprites that are just variations of the first one, so you can just have one sprite that covers them all. GAGS also allows for sprites to be recolored mid-game this is how the color customizable items from vendors work, these include: bandana, beret and shoes.

Here's an example: For bandanas you would need a separate sprite for all of these
image

but instead you just need these sprites and can assemble the layers in different ways and different colors:
image
image
image



Here's a guide on how to do it: https://hackmd.io/@tgstation/GAGS-Walkthrough

**Comment by Comxy** [comment]
GAGS seem like a cool idea, but I will pass on it for now. Maybe someone else will do it or I'll revisit this idea but I'll see if this PR gets merged first as it is right now.

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 16]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
+				var/obj/item/knife/shiv/shiv = new /obj/item/knife/shiv
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:372]: Just have the shard have a var for what it should turn into when you apply cloth, and then we don't need to have all this repeating code. do_after times can also be like that

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 17]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
-			user.put_in_hands(S)
+/obj/item/shard/attackby(obj/item/item, mob/user, params)
+	if(istype(item, /obj/item/lightreplacer))
+		var/obj/item/lightreplacer/lightreplacer = item
+		lightreplacer.attackby(src, user)
+	else if(istype(item, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/cloth = item
+		to_chat(user, span_notice("You begin to wrap the [cloth] around the [src]..."))
+		if(istype(src, /obj/item/shard/plasma))
+			if(do_after(user, 140, target = src))
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:342]:
```suggestion
			if(do_after(user, 14 SECONDS, target = src))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 18]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				var/obj/item/knife/shiv/plasma/shiv = new /obj/item/knife/shiv/plasma
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/titanium))
+			if(do_after(user, 70, target = src))
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:350]:
```suggestion
			if(do_after(user, 7 SECONDS, target = src))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 19]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				var/obj/item/knife/shiv/titanium/shiv = new /obj/item/knife/shiv/titanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else if(istype(src, /obj/item/shard/plastitanium))
+			if(do_after(user, 70, target = src))
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:358]:
```suggestion
			if(do_after(user, 7 SECONDS, target = src))
```

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 20]
```diff
@@ -332,20 +331,45 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
...
+				var/obj/item/knife/shiv/plastitanium/shiv = new /obj/item/knife/shiv/plastitanium
+				cloth.use(1)
+				to_chat(user, span_notice("You wrap the [cloth] around the [src], forming a makeshift weapon."))
+				remove_item_from_storage(src)
+				qdel(src)
+				user.put_in_hands(shiv)
+		else
+			if(do_after(user, 35, target = src))
```
Comment (tralezab) [inline code/game/objects/items/stacks/sheets/glass.dm:366]:
```suggestion
			if(do_after(user, 3.5 SECONDS, target = src))
```

**Review by tralezab — CHANGES_REQUESTED** [review]
okay lets go on to actual code stuffs

**Review by tralezab — APPROVED** [review]
awesome possum

**Pull request merged**